### ISSUE TITLE
Distinguish selected element vs. hover

### ISSUE THREAD

**Issue by davegoldin** [bpmn-io/bpmn-js#1616]
__Is your feature request related to a problem? Please describe.__

When you create an element or select an existing element, a blue dashed border appears around the element. If you click the element again (which unselects the element), the blue dashed border remains if your cursor is hovered over the element. And if an element is selected, but you hover your cursor over another element, they will both have blue dashed borders. For a new user, this could be slightly confusing. Thanks!

__Describe the solution you'd like__

One solution would be to distinguish selected vs. hover. Hover would remain the blue dashed border, selected could be a more prominent blue solid border:

```
.djs-element.selected .djs-outline {
     stroke-width: 5px;
     stroke-dasharray: none;
     }
```

__Describe alternatives you've considered__

Another option could be to slightly darken the opacity of the canvas and everything in it – except for the selected item and its sidebar – when an item is selected. That would effectively highlight the element.

**Comment by nikku** [comment]
Could you show a screen capture of the fix you propose? Maybe it is something that is worth adopting.

CC @andreasgeier

**Comment by davegoldin** [comment]
The css I posted above produces this styling. My cursor (while it did not render in the snapshot) is hovered over "hovered action." Thanks for considering.

![selected-action-solid-blue-border](https://user-images.githubusercontent.com/98906863/158940989-9576b0e9-4b4e-4304-8a81-2cf77af613da.png)

**Comment by andreasgeier** [comment]
Distinguishing hover state vs selection is definitely a topic for improvement. There would be more things to consider here, but switching to a solid line is a good first step.

I disagree with the stroke-width of 5px and would suggest 2px instead. That results in:

![Screen Shot 2022-03-18 at 08 49 31](https://user-images.githubusercontent.com/938244/158958845-b9838d79-7c30-40f0-853b-cfb768a4532a.png)

This is a super easy fix and I support this request. 

@christian-konrad FYI

**Comment by currandwyer** [comment]
@christian-konrad @andreasgeier @nikku I fully agree. As discussed with each of you, here is my initial exploration of these possibilities. I would be more than happy to double down on this and create a design specification for us to discuss. [Figma link](https://www.figma.com/file/lGwe6L9OaXpsRNpMu844WU/BPMN-diagram-js?node-id=110%3A4243)
![Screen Shot 2022-03-18 at 09 34 42](https://user-images.githubusercontent.com/11223953/158965650-cc7b2aef-c7d0-4df1-aef8-670ce4504d46.png)

**Comment by christian-konrad** [comment]
@andreasgeier  and @currandwyer  please talk to each other on this subject. It seems like you are thinking in a similar  direction.

@currandwyer's proposals in particular will have a great impact on the "finished" look of the models and reduce the cognitive and visual load during modeling.

I put this on the roadmap to considered as a later step, but keeping this open for discussion.
@currandwyer What would help is a small estimate on this (on all your proposals, not just the hover one): Will this be CSS only and therefore cheap, or is svg rendering affected? Are there any dependencies in embedding clients (e.g., the cockpit or optimize which render stuff on top off the canvas)

In case of doing this, I would like to iterate fast, put also deliver a full package of improved modeling experience.

**Comment by currandwyer** [comment]
@christian-konrad I have had some discussions with @nikku on the topic, months ago. I think it is definitely doable within a reasonable scope, but @nikku is the absolute expert on this topic. I suggest considering potentially including hover and selected states for connecting lines in the scope. I will discuss with @andreasgeier next week
![Connecting lines - hover states](https://user-images.githubusercontent.com/11223953/159004412-07946260-be2e-4ebf-8837-8bfeaf3a53b1.png)
.

**Comment by rpkoller** [comment]
i got aware of this issue over in the eca channel on the drupal slack. first a +1 for the initial suggestion to provide a way to distinguish visually  in between the selected and hover state.  and i also like the suggestion in the last comment by @currandwyer to increase the size of the handles. in the current version they are really hard to notice. 
in case the dotted line style will be kept for the hover state wcag success criterion 2.4.7 has to be kept in mind. for solid lines of 1 or 2px the used color `hsl(205, 100%, 50%)` works out with a contrast of `3,1:1` against the white background (for non-text contrasts a ration of `3:1` is enough according to SC 1.4.11). but with dotted lines the gaps in the line decrease the focus indication area - the outline therefor is hard to see. if the dotted style will be kept for the hover state then maybe increasing the width to perhaps at least `3px` might be an option (that is something that got suggested in the following wcag21 discussion on github: https://github.com/w3c/wcag21/issues/424 

> A dotted or dashed line can be used if at least 3px in thickness.

**Comment by nikku** [comment]
Moving to backlog for the time being. Thanks everyone for the input.

**Comment by christian-konrad** [comment]
Not distinguishing both states makes multi-select somehow weird.
We should address this before we build the multi-select context pad.
Linking it to the epic.
https://user-images.githubusercontent.com/56470362/166655292-7742ccb8-1c37-4d85-a734-c7912cb7f629.mov

**Comment by nikku** [comment]
@currandwyer Could you extract [your suggestion on improved hover / select states for connections](https://github.com/bpmn-io/bpmn-js/issues/1616#issuecomment-1072372003) into a separate issue.

We'll tackle this in a simple, straight forward manner as [suggested by you and @andreasgeier](https://github.com/bpmn-io/bpmn-js/issues/1616#issuecomment-1072111300). Everything beyond that I'll treat as out of scope for the moment.

**Comment by christian-konrad** [comment]
@nikku please tell me what's the scope in your opinion for now.

I agree to put the connections into a seperate issue.

But what are you now aiming to do on the node hover + select state?

**Comment by nikku** [comment]
@currandwyer Also consider adding your proposal on "improved selection box" (with round corners for the select / hover) as an additional feature request. It will be quite some work to get that incorporated I think and if we want it we'd need to tackle it appropriately (i.e. answer how it would work as a generic diagram concern.

**Comment by nikku** [comment]
https://github.com/bpmn-io/diagram-js/pull/640 proposes a straight forward improvement (solid selection outline).

**Comment by nikku** [comment]
Closed via https://github.com/bpmn-io/bpmn-js/commit/b158addcc8f46ecbc9210b6b569823686547a323.
